scripts/cloudPBFShardControl/pbfReShard.py

```diff
@@ -0,0 +1,315 @@
+        """
+        for child_shard in shard.children:
+            final_release_dir = os.path.join(self.releasePbfDir,
+                                             str(child_shard.tile_z), str(child_shard.tile_x), str(child_shard.tile_y))
+            if not os.path.exists(final_release_dir):
+                os.makedirs(final_release_dir)
+            # copy generated shards to pbf release folder
+            pbf_copy_cmd = ["cp", self.tmpDir + child_shard.pbf_file_name(), final_release_dir]
+            logger.info("copy %s to %s.", child_shard.pbf_file_name(), final_release_dir)
+            subprocess.run(pbf_copy_cmd, stdout=subprocess.PIPE, text=True)
+        # delete oversize shard form pbf release folder
+        logger.info("delete shard %s from %s.", shard.pbf_file_name(),
+                    os.path.join(self.releasePbfDir, str(shard.tile_z),
+                                 str(shard.tile_x), str(shard.tile_y)))
+        shutil.rmtree(os.path.join(self.releasePbfDir, str(shard.tile_z), str(shard.tile_x), str(shard.tile_y)))
+
+    def update_sharding_tree(self, shard: "SlippyTileQuadTreeNode"):
+        """
+        update sharding quad tree content according to sharding extract
+        :param shard: SlippyTileQuadTreeNode object
+        """
+        # Handle last line to prevent IndexError
+        if shard.name() + "\n" in self.shardingContent[-1] and "+" not in self.shardingContent[-1]:
+            print(f"Found {shard.name()} in last line: {self.shardingContent[-1]}")
+            self.shardingContent[-1] = shard.name() + "+\n"
+            for child in shard.children:
+                self.shardingContent.append(f"{child.tile_z}-{child.tile_x}-{shard.tile_y}\n")
+            self.change = True
+        else:
+            for index, line in enumerate(self.shardingContent):
+                if shard.name() + "\n" in line and not shard.is_leaf():
+                    logger.info("found %s in line %s. updating sharding tree", shard.name(), index)
+                    self.shardingContent[index] = shard.name() + "+\n"
+                    for child in shard.children:
+                        self.shardingContent.append(f"{child.tile_z}-{child.tile_x}-{shard.tile_y}\n")
+                    self.change = True
+                    break
+
+    def apply_sharding_changes(self):
+        """
+        create new sharding.txt file
+        """
+        if self.change:
+            logger.info("creating new sharding tree file")
+            shard_date = date.today().strftime("%Y%m%d")
+            new_sharding_quadtree_file = f"sharding_quadtree_{shard_date}.txt"
+            with open(os.path.join(self.tmpDir, new_sharding_quadtree_file), 'w') as output_file:
+                output_file.writelines(self.shardingContent)
+            # copy original sharding file
+            copy_command(self.shardingFileName, os.path.join(self.releasePbfDir, f"sharding_original_{shard_date}.txt"))
```

Comment:
Three things:

1. Don't you want to use the filename that was passed in here?
2. we need to add the hour and minute to this date string.
3. I would tack the data on the end of the filename.

In other words, if the user passes in:
`special_sharding_file.txt`
then we want the script to copy it to:
`special_sharding_file.txt.202206100327`

Reply:
resolved.